### PR TITLE
feat: create reusable popup

### LINKED ISSUE #96: [Task]: Create reusable popup component
**Task description.**

Create reusable popup container that can be opened from anywhere and overlays on top of everything
image

**No need to do the creation form now**, that will come in a different task

**Requirements.**
The popup should have the following parts
- Header: fixed on top of the popup
- Body: part of the popup which can be scrollable in case the content doesnt fit
- Footer: fixed at the bottom

### PR DESCRIPTION
**Pull request by evavirseda**
Fixes #96 

## Changes

  - Create reusable popup component
  - Add an example
  
### Checks

* [x] Have you catchup your branch with the latest state in base?
* [x] Have you lint your code locally prior to submission?
* [x] Have you reviewed your proposed changes and removed debris?
* [x] Have you successfully ran due tests with your changes?

### CODE REVIEW

File: components/Popup.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Dispatch, FC, SetStateAction, useState } from 'react'
+import { Box } from '@primer/react'
+import { Dialog } from '@primer/react/drafts'
+
+const Modal: FC<{
```
Comment (cpl121) [inline components/Popup.tsx:5]: The name has changed to `Popup`

**Review by cpl121 — APPROVED** [review]
LGTM! 💯

File: components/Popup.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { Dispatch, FC, SetStateAction, useState } from 'react'
```
Comment (VmMad) [inline components/Popup.tsx:1]: useState is unused

File: components/Popup.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { Box } from '@primer/react'
+import { Dialog } from '@primer/react/drafts'
```
Comment (VmMad) [inline components/Popup.tsx:3]: We have a problem here. You are using a draft.
These are two different Dialogs
```suggestion
import { Dialog } from '@primer/react'
```

File: components/Popup.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+const Popup: FC<{
+    isOpen: boolean
+    setIsOpen: Dispatch<SetStateAction<boolean>>
+    children: any
+    title: string
+    onClick: () => void
+}> = ({ isOpen, setIsOpen, children, title, onClick }) => {
+    return (
+        <>
+            {isOpen && (
```
Comment (VmMad) [inline components/Popup.tsx:14]: With the correct Dialog this condition isn't needed anymore

File: components/Popup.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                <Dialog
```
Comment (VmMad) [inline components/Popup.tsx:15]: The code would look something like this: ( It isn't completed, just a test ) 

```<>
            <Dialog
                title={title}
                isOpen={isOpen}
                onDismiss={() => setIsOpen(false)}
                aria-labelledby='label'
                sx={{ height: '100%', overflowY: 'scroll' }}
            >
                <Box p={3}>{children}</Box>
                <Box display='flex' mt={3} justifyContent='flex-end'>
                    <Button
                        onClick={() => {
                            onClick(), setIsOpen(false)
                        }}
                    >
                        Ok
                    </Button>
                </Box>
            </Dialog>
        </>

**Review by VmMad — CHANGES_REQUESTED** [review]
Great work. Let me know what you think about these comments

**Review by VmMad — APPROVED** [review]
:1st_place_medal:

**Pull request merged**